### PR TITLE
grant user/role; closes #180

### LINKED ISSUE #180: [server] please grant user/role

### PR DESCRIPTION
**Pull request by jonnymbgood**

### CODE REVIEW

File: internal/server/action/grantRole.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		roleObj, ok := root.Roles[targetRole]
+		if !ok {
+			r.Error = errors.New("Role does not exist")
+			return
+		}
+
+		// Make sure the permission isn't already associated
+		exists := false
+		for _, permission := range roleObj.Permissions {
+			if permission.Action == targetPermission.Action && permission.TargetType == targetPermission.TargetType && permission.TargetID == targetPermission.TargetID {
```
Comment (thuey) [inline internal/server/action/grantRole.go]: Do we want to bother handling the case in a special way if, for example, the command is grant permission "check k8s prod", but they already have "check k8s *"? How about on revoke?

Comment (jonnymbgood) [reply internal/server/action/grantRole.go]: We could, but what would we do when the reverse happens (i.e. they have `"check k8s prod"` and then grant `"check k8s *"`? I guess I would be ok with a warning, but I think we may want to grant it even if it doesn't make sense. Especially if they are trying to downgrade without locking someone out (add the specific permissions, then remove the non-specific permissions).

Comment (jonnymbgood) [reply internal/server/action/grantRole.go]: If we want to issue a warning we can handle that in another issue. Do you think it would be worth doing?

**Comment by thuey** [comment]
It's one of those situations, where the desired behavior from the user's
perspective is a bit ambiguous. Are those separate permissions that can be
independently granted and revoked? Or does the user expect us to coalesce
them into a single permission? I agree that this could be handled in a
separate issue, but I do think it needs to be addressed somehow. Here are a
few approaches I think we could take:

1. Inform: We decide to either treat them as separate or coalesce them, but
inform the user what happened.
2. Warn: We decide to either treat them as separate or coalesce them, but
we require the user to add the equivalent of a "force" flag to proceed.
3. Error: We show the user an error when there are overlapping permissions
and require them to manually resolve issues. For example, if the grantee
already has "check k8s prod" and the grantor was trying to grant "check k8s
*", the grantor would have to remove "check k8s prod" before they could run
"check k8s *"

I don't think 3 is very user friendly. I'm split between 1 or 2.

What do you think/can you think of other options?

**Comment by jonnymbgood** [comment]
I would go with option 1. Our permission algorithm is denied unless explicitly granted. If more than 1 permission matches that's ok, since we don't coalesce permissions. I would rather treat each permission entry as it's own entity, rather than modifying an abstract "permissions set".

**Comment by thuey** [comment]
Sounds good. I do feel it's important to warn them if they revoke a
permission but another permission still gives the user/role access to what
they revoked

**Comment by jonnymbgood** [comment]
Cool. Can you create an issue to that effect with the scenario in your head and a proposed warning message? I'm guessing we want to do this for both grant and revoke.

**Comment by thuey** [comment]
Yep.

**Comment by jonnymbgood** [comment]
Thanks!

**Pull request merged**